core/trino-main/src/main/java/io/trino/operator/Driver.java

```diff
@@ -179,6 +179,12 @@ public void close()
             return;
         }
 
+        for (Operator operator : allOperators) {
```

Comment:
I believe there can be a race condition here between creating page source and cancelling it.